PowerEditor/src/MISC/Common/Common.cpp

```diff
@@ -1329,25 +1329,112 @@ int nbDigitsFromNbLines(size_t nbLines)
 	return nbDigits;
 }
 
+namespace
+{
+	constexpr TCHAR timeFmtEscapeChar = 0x1;
+	constexpr TCHAR middayFormat[] = _T("tt");
+
+	// Returns AM/PM string defined by the system locale for the specified time.
+	// This string may be empty or customized.
+	generic_string getMiddayString(const TCHAR* localeName, const SYSTEMTIME& st)
+	{
+		generic_string midday;
+		// Obtain the length of the resulting string first.
+		int ret = GetTimeFormatEx(localeName, 0, &st, middayFormat, nullptr, 0);
+		if (ret > 0)
+		{
+			midday.resize(ret);
+			ret = GetTimeFormatEx(localeName, 0, &st, middayFormat, &midday[0], static_cast<int>(midday.size()));
+			if (ret != 0)
+				midday.resize(ret - 1); // Remove the null-terminator.

```

Comment:
Why bother to remove null terminator?

Reply:
Because string algorithms such as `std::string::erase` and `std::string::insert` will lead to incorrect results otherwise.
It had actually happened to me until I figured out.